diffkemp/simpll/DifferentialFunctionComparator.cpp

```diff
@@ -1112,15 +1124,15 @@ int DifferentialFunctionComparator::cmpBasicBlocks(
 
             // Try to find a match by moving one of the instruction iterators
             // forward (find a code relocation).
-            if (config.Patterns.Relocations
+            if (config.Patterns.Relocations && !suppressRelocationsAndSMT
```

Comment:
The `suppressRelocationsAndSMT` parameter feels a bit clumsy, TBH. What if we find out that we need to turn off more builtin patterns in future?

Two alternatives come to mind:
1. Turn off `config.Patterns.Relocations` and `config.UseSMT` for the particular call of `cmpBasicBlocksFromInstructions`.
2. Make `SMTComparator` use a separate instance of `DifferentialFunctionComparator` having a different config.

Reply:
Yeah, I wasn't a big fan of this either but couldn't think of a better solution. The first alternative feels quite dirty as well, `Config` represents the user configuration and should therefore IMO be constant throughout the whole run. Furthermore, I am not sure if it is even possible to use it, `cmpBasicBlocks` is declared as const inside LLVM API, i.e., it cannot change the values of data fields. So even if we changed `config` to non-const, it probably wouldn't be possible to modify its fields inside `SMTBlockComparator` since it is called from `cmpBasicBlocks`.

I can try the second alternative but there may be other drawbacks. One thing that comes to mind -- `DifferentialFunctionComparator` has some state, e.g. some patterns keep track of various information. We would somehow have to transfer this information (the current constructor is rather basic and accepts only a few arguments). Maintaing the list of fields that need to be transferred for the new `DifferentialFunctionComparator` instance to be fully functional seems quite tedious. WDYT?